### PR TITLE
Allow Broker Platform Credential Rotation during full resync in K8S platforms

### PR DESCRIPTION
**Pull request by NickyMateev**
# Allow Broker Platform Credential Rotation during full resync in K8S platforms

## Motivation

Unlike CF installations, K8S clusters are very volatile environments and it's easier to delete broker resources there. In case a broker is deleted in K8S directly (or CF for that matter), the broker record and its broker platform credentials will still exist in SM. And whenever the K8S agent retriggers full resync it would fail to register the broker back in the platform as it would try to essentially rotate the broker platform credentials - something we do not allow. We only allow full resync to initially set broker platform credentials, but now we will allow this rotation for K8S platforms,

## Approach

Make broker platform credentials controller less restrictive.

## Pull Request status

* [x] Initial implementation
* [x] Integration tests

### CODE REVIEW

File: test/osb_test/catalog_test.go [hunk 1]
```diff
@@ -413,11 +413,37 @@ var _ = Describe("Catalog", func() {
 			})
 
 			Context("when broker platform credentials change out of notification processing context when already exist", func() {
-				It("should return 409", func() {
+				FIt("should return 409 for non-kubernetes platforms", func() {
```
Comment (pankrator) [inline test/osb_test/catalog_test.go:416]: Remove the focus on the test

Comment (DimitarPetrov) [inline test/osb_test/catalog_test.go:416]: `FIt` Probably leftover

File: test/osb_test/catalog_test.go [hunk 2]
```diff
@@ -413,11 +413,37 @@ var _ = Describe("Catalog", func() {
...
+					By("initial K8S credentials should work")
+					k8sOSBClient.GET(osbURL + "/v2/catalog").
+						Expect().Status(http.StatusOK).JSON().Object().ContainsKey("services")
+
+					newUsername, newPassword = test.RegisterBrokerPlatformCredentialsExpect(k8sPlatformClient, prefixedBrokerID, http.StatusOK)
+					k8sOSBClient.SetBasicCredentials(ctx, newUsername, newPassword)
+
+					By("rotation of K8S credentials broker platform credentials should work")
+					k8sOSBClient.GET(osbURL + "/v2/catalog").
+						Expect().Status(http.StatusOK).JSON().Object().ContainsKey("services")
```
Comment (pankrator) [inline test/osb_test/catalog_test.go:446]: Will this be enough to verify that the credentials are rotated?

Comment (NickyMateev) [reply test/osb_test/catalog_test.go:446]: discussed

**Pull request merged**